jujugui/static/gui/src/app/app.js

```diff
@@ -491,7 +491,7 @@ YUI.add('juju-gui', function(Y) {
       let password = null;
       let macaroons = null;
       let credentials = this.user.controller;
-      if (credentials.areAvailable) {
+      if (credentials && credentials.areAvailable) {
```

Comment:
I understand that, but I am still confused about why in that case user.controller would be null or undefined. It should still be a property right? What I actually encounter when just visiting the python server set up by "make run" is another error: Cannot read property 'bakeryFactory' of undefined.